src/plugins/vis_augmenter/public/vega/helpers.ts

```diff
@@ -340,3 +345,43 @@ export const addPointInTimeEventsLayersToSpec = (
 
   return newSpec;
 };
+
+/**
+ * Interaction handling functions mapped by their action names.
+ */
+export const interactionHandlersByAction = {
+  [VisInteractionHandler.HANDLE_POINT_IN_TIME_CLICK]: (
+    _event: ScenegraphEvent,
+    item?: Item | null
+  ) => {
+    if (isPointInTimeAnnotation(item)) {
+      // TODO: Show the events flyout
+    }
+  },
+};
+
+/**
+ * Updating the vega-lite spec to add interaction config that is used to add event handlers on the visualization view.
+ * Note: Since, we cannot have handler functions added directly to the spec object as it is stringified,
+ * we pass the handler names as part of the config and use the @see interactionHandlersByAction to get the corresponding
+ * handler when required.
+ */
+export const addPointInTimeInteractionsConfig = (config: object) => {
+  const kibana = get(config, 'kibana', {});
+  const withInteractionsConfig = {
+    ...kibana,
+    visInteractions: [
+      ...(kibana.visInteractions || []),
+      { event: 'click', handlerName: VisInteractionHandler.HANDLE_POINT_IN_TIME_CLICK },
+    ],
+  };
+
+  return {
+    ...config,
+    kibana: withInteractionsConfig,
+  };
+};
+
+export const isPointInTimeAnnotation = (item?: Item | null) => {
+  return item?.datum?.userAction === VisInteraction.VIEW_EVENTS_FLYOUT;
+};
```

Comment:
There's a naming mismatch here - the function name implies that it's checking if the item is a particular type (point in time annotation), whereas the actual check is whether the userAction is the view events flyout. Either we're checking the wrong thing or the function is named the wrong thing.

Reply:
Updated logic to check for the annotation type